compiler/src/codegen/compcore.re

```diff
@@ -2624,173 +3699,152 @@ let compile_exports = (env, {functions, imports, exports, num_globals}) => {
-              ),
-            ),
-        }),
-        add_dummy_loc({
-          Ast.name: encode_string("memory"),
-          Ast.edesc:
-            add_dummy_loc(
-              Ast.MemoryExport(add_dummy_loc @@ Int32.of_int(0)),
-            ),
-        }),
-      ],
-    ),
+  ignore @@ List.mapi(compile_getter, exports);
+  ignore @@ Export.add_function_export(wasm_mod, "_start", "_start");
+  ignore @@
+  Export.add_function_export(
+    wasm_mod,
+    Printf.sprintf("func_%d", cleanup_globals_idx),
+    "GRAIN$CLEANUP_GLOBALS",
+  );
+  ignore @@
+  Export.add_global_export(
+    wasm_mod,
+    Printf.sprintf("global_%d", num_globals + 1),
+    Ident.name(table_size),
   );
 };
 
-let compile_tables = (env, prog) =>
-  /*let table_size = compute_table_size env prog in*/
-  [];
-    /*add_dummy_loc {
-        Ast.ttype=Types.TableType({
-            Types.min=Int32.of_int table_size;
-            Types.max=None;
-          }, Types.FuncRefType)
-      };*/
-
-let compile_elems = (env, prog) => {
+let compile_tables = (wasm_mod, env, {functions, imports} as prog) => {
   let table_size = compute_table_size(env, prog);
-  Wasm.Ast.[
-    add_dummy_loc({
-      index: add_dummy_loc(Int32.zero),
-      offset:
-        add_dummy_loc([
-          add_dummy_loc(Ast.GlobalGet(add_dummy_loc @@ Int32.of_int(0))),
-        ]),
-      init: BatList.init(table_size, n => add_dummy_loc(Int32.of_int(n))),
-    }),
-  ];
+  let import_names =
+    List.filter_map(
+      ({mimp_kind, mimp_type, mimp_mod, mimp_name}) =>
+        switch (mimp_kind, mimp_type) {
+        | (MImportGrain, _)
+        | (_, MFuncImport(_)) =>
+          Some(get_imported_name(mimp_mod, mimp_name))
+        | _ => None
+        },
+      imports,
+    );
+  let function_name = i => Printf.sprintf("func_%d", i);
+  let function_names =
+    List.mapi((i, _) => function_name(env.func_offset + i), functions);
+  let all_import_names =
+    List.concat([
+      import_names,
+      function_names,
+      ["_start", function_name(table_size - 1)],
+    ]);
+  Function_table.set_function_table(
+    wasm_mod,
+    table_size,
+    max_int,
+    all_import_names,
+    Expression.global_get(
+      wasm_mod,
+      get_imported_name(runtime_mod, reloc_base),
+      Type.int32,
+    ),
+  );
 };
 
-let compile_globals = (env, {num_globals} as prog) =>
-  List.append(
-    BatList.init(1 + num_globals, _ =>
-      add_dummy_loc({
-        Ast.gtype:
-          [@implicit_arity] Types.GlobalType(Types.I32Type, Types.Mutable),
-        Ast.value:
-          add_dummy_loc([add_dummy_loc @@ Ast.Const(const_int32(0))]),
-      })
-    ),
-    [
-      add_dummy_loc({
-        Ast.gtype:
-          [@implicit_arity] Types.GlobalType(Types.I32Type, Types.Immutable),
-        Ast.value:
-          add_dummy_loc([
-            add_dummy_loc @@
-            Ast.Const(const_int32(compute_table_size(env, prog))),
-          ]),
-      }),
-    ],
+let compile_elems = (wasm_mod, env, prog) =>
```

Comment:
Does this do nothing?

Reply:
Huh, yeah, the diff makes this look really weird. But yes, this doesn't do anything (at least for now). Previously this function did what `compile_tables` does now, and the old `compile_tables` didn't do anything. There's an issue though to clean up this file, so I'll probably take care of this then.